iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp

```diff
@@ -29,16 +30,12 @@ namespace iox
 {
 namespace capro
 {
-static constexpr uint16_t InvalidID = 0u;
-static const IdString_t InvalidIDString{"0"};
-static constexpr uint16_t AnyService = 0xFFFFu;
-static constexpr uint16_t AnyInstance = 0xFFFFu;
-static constexpr uint16_t AnyEvent = 0xFFFFu;
+static const IdString_t InvalidString{"INVALID"};
 static constexpr char AnyServiceString[]{"65535"};
 static constexpr char AnyInstanceString[]{"65535"};
 static constexpr char AnyEventString[]{"65535"};
```

Comment:
Please make these a `static constexpr const char foo[]`, then `TruncateToCapacity` is not necessary when converted to an `IdString`. Maybe the this should even be an `IdString` in the first place. I guess with C++14 the `cxx::string` ctor can be made constexpr, at least for string literals ... oh and why is `AnyServiceString` `"65535"`? I guess it's a remnant of the `uint16_t`. Could this maybe also be changed to something like `*` or `#$%&@_*_@&%$#`?

Could the `String` suffix be removed here? There is now only a ctor for `IdString` so there is no ambiguity. Or maybe change it to a `ID` suffix which is shorter.

Reply:
Those are removed in #854 Happy to change the naming.